### PR TITLE
Add index put

### PR DESCRIPTION
**Pull request by HelloShirleyHello**
# Description

user wants to convert following torch script to trt engine.

```
class Net(nn.Module):
    def __init__(self):
        super(Net2, self).__init__()

    def forward(self, x):
        y = torch.zeros(1,1,224)
        x[:, 1, 1, :]=y
        return x
```
trt can't support in-place memory update right now. So I recommend  them to use "index_put" op, which will store the result in new memory.

This pull request is for support index_put in trt.  I use scatterND op in tensorrt to support index_put op in pytorch. 
IScatterLayer has three input tensors: Data D , Indices I , and Updates U.

My code include:
1. convert index_put indices to int32 type.
2. broadcast all the indices into same shape and unsqueeze last dimension.
3. concat all the indices on last dimension to get "I" for scatter layer.
4. compute the expected "U" tensor shape by "D" and "I".
5. broadcast input update tensor to "U".
6. add scatter layer when accumulate is false in index_put.
7. add gather layer and scatter layer when accumulate is true in index_put. 

Fixes # (issue)

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes 
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

**Comment by narendasan** [comment]
@HelloShirleyHello Can you setup the pre-commit system (`pre-commit --install && pre-commit run --all-files`) to apply the lint for the repo.

File: core/conversion/converters/impl/index_put.cpp [hunk 1]
```diff
@@ -0,0 +1,144 @@
+#include <ATen/ATen.h>
```
Comment (narendasan) [inline core/conversion/converters/impl/index_put.cpp:1]: Can you just add the converter to shuffle.cpp or rename this scatter.cpp?

File: core/util/trt_util.cpp [hunk 2]
```diff
@@ -6,6 +6,39 @@ namespace torch_tensorrt {
 namespace core {
 namespace util {
 
+nvinfer1::Dims broadcastDim(nvinfer1::Dims a, nvinfer1::Dims b) {
```
Comment (narendasan) [inline core/util/trt_util.cpp:9]: Can you just call this `broadcast`

File: core/conversion/converters/BUILD [hunk 3]
```diff
@@ -84,6 +84,7 @@ cc_library(
         "impl/topk.cpp",
         "impl/unary.cpp",
         "impl/unsqueeze.cpp",
+        "impl/index_put.cpp"
```
Comment (narendasan) [inline core/conversion/converters/BUILD:87]: Make sure to add this file to the `CMakeLists.txt` as well

File: core/conversion/converters/impl/index_put.cpp (hunk 1 above)
Comment (HelloShirleyHello) [reply core/conversion/converters/impl/index_put.cpp]: renamed it to scatter.cpp

File: core/util/trt_util.cpp (hunk 2 above)
Comment (HelloShirleyHello) [reply core/util/trt_util.cpp]: renamed it to broadcast

File: core/conversion/converters/BUILD (hunk 3 above)
Comment (HelloShirleyHello) [reply core/conversion/converters/BUILD]: added it in CMakeLists

**Comment by HelloShirleyHello** [comment]
> @HelloShirleyHello Can you setup the pre-commit system (`pre-commit --install && pre-commit run --all-files`) to apply the lint for the repo.

**Comment by peri044** [comment]
@HelloShirleyHello  Can you address the review comments and fix the tests ?
